### ISSUE TITLE
scala 2.13.1:getting  [error] java.lang.NoSuchMethodError: scala.tools.nsc.Global.reporter()Lscala/tools/nsc/reporters/Reporter

### ISSUE THREAD

**Issue by sitwaheed9** [sksamuel/scapegoat#447]
i am trying to use scapegoat in scala 2.13.1

i have added sbt-scapegoat plugin in project/plugin.sbt

`addSbtPlugin("com.sksamuel.scapegoat" %% "sbt-scapegoat" % "1.1.0")
`
here is my build.sbt

```
lazy val scala212 = "2.12.8"

lazy val scala213 = "2.13.1"

scalaVersion in ThisBuild := scala213

lazy val supportedScalaVersions = List(scala213, scala212)

crossScalaVersions := supportedScalaVersions
 
scapegoatVersion in ThisBuild := "1.3.11"

libraryDependencies := {
  val VersionSchema = """(\d+)\.(\d+)\.(\d+)""".r
  scalaVersion.value match {
    case VersionSchema("2", "12", "8") =>
      libraryDependencies.value ++ Seq(
      "ch.qos.logback" % "logback-classic" % "1.2.3",
      "org.scalatest" %% "scalatest" % "3.1.0" % Test,
      "io.gatling.highcharts" % "gatling-charts-highcharts" % "3.3.1" % "test",
      "io.gatling" % "gatling-test-framework" % "3.3.1" % "test")
    case VersionSchema("2", "13", "1") =>
  libraryDependencies.value ++ Seq(
 "co.pragmati" %% "swagger-ui-akka-http" % "1.3.0"
  ,"com.sksamuel.scapegoat" %% "scalac-scapegoat-plugin"%"1.3.11"
  )
  }
  }

```   
when i run the scapegoat command in my sbt i am getting following exception

```
[error] ## Exception when compiling 46 sources to /home/sara/git/gitinterpret/interpret/target/scala-2.13/scapegoat-classes
[error] scala.tools.nsc.Global.reporter()Lscala/tools/nsc/reporters/Reporter;
[error] com.sksamuel.scapegoat.ScapegoatComponent$$anon$1.run(plugin.scala:172)
[error] scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1506)
[error] scala.tools.nsc.Global$Run.compileUnits(Global.scala:1490)
[error] scala.tools.nsc.Global$Run.compileSources(Global.scala:1482)
[error] scala.tools.nsc.Global$Run.compile(Global.scala:1614)
[error] xsbt.CachedCompiler0.run(CompilerInterface.scala:130)
[error] xsbt.CachedCompiler0.run(CompilerInterface.scala:105)
[error] xsbt.CompilerInterface.run(CompilerInterface.scala:31)
[error] java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error] java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error] java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error] java.base/java.lang.reflect.Method.invoke(Method.java:566)
[error] sbt.internal.inc.AnalyzingCompiler.call(AnalyzingCompiler.scala:237)
[error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:111)
[error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:90)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$3(MixedAnalyzingCompiler.scala:82)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
[error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:133)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:73)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:116)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:307)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:307)
[error] sbt.internal.inc.Incremental$.doCompile(Incremental.scala:106)
[error] sbt.internal.inc.Incremental$.$anonfun$compile$4(Incremental.scala:87)
[error] sbt.internal.inc.IncrementalCommon.recompileClasses(IncrementalCommon.scala:116)
[error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:63)
[error] sbt.internal.inc.Incremental$.$anonfun$compile$3(Incremental.scala:89)
[error] sbt.internal.inc.Incremental$.manageClassfiles(Incremental.scala:134)
[error] sbt.internal.inc.Incremental$.compile(Incremental.scala:80)
[error] sbt.internal.inc.IncrementalCompile$.apply(Compile.scala:67)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:311)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:269)
[error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCom
```

how can i resolve this error?

**Comment by mccartney** [comment]
You seem to be using some old version of the plugin - i.e. `1.3.11`, while the current one is `1.4.6`.

Also, as of now we are publishing builds only for `2.13.2` and `2.13.3` and you seem to be using `2.13.1`. And the versions are not binary compatible between even minor Scala versions (with symptoms like you observe). Thus you have at least two options:
- switch your Scala language library to `2.13.3` and use `scalac-scapegoat-plugin_2.13.3` version `1.4.6`
- stay with Scala `2.13.1` and then use version `1.4.4` of the plugin (which was the last one we published for `2.13.1`)

**Comment by mccartney** [comment]
No response received. I still think what I have described in my last comment holds.
